### PR TITLE
Fix oslc assertion on type constructor for unknown struct

### PR DESCRIPTION
**Pull request by lgritz**
The assertion was not needed, errors issue before this code is reached.

### CODE REVIEW

File: src/liboslcomp/typecheck.cpp [hunk 1]
```diff
@@ -2272,8 +2272,11 @@ OSLCompilerImpl::code_from_type (TypeSpec type) const
             out = 's';
         else if (elem == TypeDesc::NONE)
             out = 'x';
-        else
-            ASSERT (0);
+        else {
+            // error(ustring(), 0, "Possible compiler bug: %s(%s)",
```
Comment (fpsunflower) [inline src/liboslcomp/typecheck.cpp]: Remove this block of commented out code? (and maybe leave a comment at the end about how an empty string being returned signals an unknown type?)

Comment (lgritz) [reply src/liboslcomp/typecheck.cpp]: Updated.

Comment (lgritz) [reply src/liboslcomp/typecheck.cpp]: OK with this the latest update here?

**Comment by fpsunflower** [comment]
LGTM!

**Pull request merged**